simulation_ws/src/aws_robomaker_simulation_common/nodes/route_manager

```diff
@@ -48,55 +200,58 @@ class RouteManager():
     
     '''
 
-    # return an iterator over the goals
-    route_modes = {
-        'inorder': lambda goals: itertools.cycle(goals),
-        'random' : lambda goals: (random.choice(goals) for i in itertools.count()),
-    }
-
     def __init__(self):
         self.route = []
 
         self.client = actionlib.SimpleActionClient('move_base', MoveBaseAction)
         self.client.wait_for_server()
 
-        self.route_mode = rospy.get_param('~mode')
-        if self.route_mode not in RouteManager.route_modes:
-            rospy.logerr("Route mode '%s' unknown, exiting route manager", self.route_mode)
-            return
-
-        poses = rospy.get_param('~poses')
-        if not poses:
-            rospy.loginfo("Route manager initialized no goals, unable to route")
+        self.map = GoalGenerator()
+        rospy.loginfo("Route manager initialized with random goal generator")
 
-        self.goals = RouteManager.route_modes[self.route_mode](poses)
-        rospy.loginfo("Route manager initialized with %s goals in %s mode", len(poses), self.route_mode)
+        self.bad_goal_counter = 0
 
     def to_move_goal(self, pose):
+        if pose is None:
+            raise ValueError("Goal position cannot be NULL")
+
         goal = MoveBaseGoal()
         goal.target_pose.header.stamp = rospy.Time.now()
         goal.target_pose.header.frame_id = "map"
-        goal.target_pose.pose.position = Point(**pose['pose']['position'])
-        goal.target_pose.pose.orientation = Quaternion(**pose['pose']['orientation'])
+        goal.target_pose.pose.position = pose.position
+        goal.target_pose.pose.orientation = pose.orientation
         return goal
 
+
     def route_forever(self):
         rate = rospy.Rate(1)
         while not rospy.is_shutdown():
             try:
-                rospy.loginfo("Route mode is '%s', getting next goal", self.route_mode)
-                current_goal = self.to_move_goal(next(self.goals))
+                if self.bad_goal_counter > 10:
+                    raise ValueError("Bad goal limit exceeded")
+
+                rospy.loginfo("Quering goal generator for the next goal")
+                current_goal = self.to_move_goal(self.map.get_valid_random_world_pos())
                 rospy.loginfo("Sending target goal: %s", current_goal)
                 self.client.send_goal(current_goal)
-
-                if not self.client.wait_for_result():
-                    rospy.logerr("Move server not ready, will try again...")
-                else:
-                    if self.client.get_result():
-                        rospy.loginfo("Goal done: %s", current_goal)
+                
+                try:
+                    #wait 5sec for global plan to be published. If not, scan for a new goal..
```

Comment:
Why is this necessary? Was something breaking?

Reply:
this is the best sanity check to see if the navigation stack found a global plan to the goal, hence making the goal "valid"